docs/proposal-006-build-improvements.md

```diff
@@ -0,0 +1,55 @@
+# Design Proposal #006: Build Improvements
+
+## Current State
+
+At the start of the project we've picked CircleCI as it appears to be a default choice at Weaveworks.
+
+The original solution involved:
+- running `docker build` to create a build image
+- use build image to run tests produce final image
+
+Publishing build and final images hasn't been automated, but some users requested it.
+
+When we switched [to modules](https://github.com/weaveworks/eksctl/pull/917), we have increased the
+compexity of how build image worked. It's hard for a new person to understand it, e.g. for a new
```

Comment:
why is it hard, what is required now that was not there before (brief explanation, doesn't need to go into deep details)

Reply:
It's just complicated. Try reading what we have.

https://github.com/weaveworks/eksctl/blob/950f9bc695234107725234b9e1a9c9d2ee54e51f/.circleci/config.yml#L3-L38

https://github.com/weaveworks/eksctl/blob/950f9bc695234107725234b9e1a9c9d2ee54e51f/Makefile#L188-L204


And what becomes of that section of `.circleci/config.yml` on this branch:

https://github.com/weaveworks/eksctl/blob/2bae42e5bcd8991ed734851144b12e2bd5af68a8/.circleci/config.yml#L3-L16

Which now points you at much easier to understand make target:

https://github.com/weaveworks/eksctl/blob/2bae42e5bcd8991ed734851144b12e2bd5af68a8/Makefile#L68-L73

There is still some complexity in how the build image is put together, but that's in a separate [`Makefile.docker`](https://github.com/weaveworks/eksctl/blob/2bae42e5bcd8991ed734851144b12e2bd5af68a8/Makefile.docker) and you don't have to dig into it just to figure out why your tests may have failed in CI and passed locally.

Reply:
Could you reflect this in the document and resolve this comments? Sounds very good.